tests/ai/test_databricks_client.py

```diff
@@ -739,3 +766,33 @@ def mock_execute_statement(
         client._execute_uc_function(
             good_function_info, {EXECUTE_FUNCTION_ARG_NAME: {"invalid_param": "a"}}
         )
+
+
+@requires_databricks
+def test_extra_params_when_executing_function_e2e(client: DatabricksFunctionClient, monkeypatch):
+    monkeypatch.setenv(UNITYCATALOG_AI_CLIENT_EXECUTION_TIMEOUT, "5")
+    with generate_func_name_and_cleanup(client) as func_name:
+        full_func_name = f"{CATALOG}.{SCHEMA}.{func_name}"
+        sql_body = f"""CREATE FUNCTION {full_func_name}()
+RETURNS STRING
+LANGUAGE PYTHON
+AS $$
+    import time
+
+    time.sleep(100)
+    return "10"
+$$
+"""
+        client.create_function(sql_function_body=sql_body)
+        time1 = time.time()
+        # default wait_timeout is 30s
+        client.execute_function(full_func_name)
+        time_total1 = time.time() - time1
```

Comment:
```suggestion
        time_total1 = time.time() - time1
        assert 29 <= time_total1 <= 31
```

Is this more straightfoward?

Reply:
This is actually not true, because the total execution time include wait_timeout + retry time + data parsing time

Reply:
I'm just not big fan of `assert abs(abs(time_total2 - time_total1) - 20) < 5`. I'm fine if that's not flaky.

Reply:
I think 5 seconds is enough buffer for 'flaky' tolerance :) We'll see how it goes!